src/opentimelineio/anyDictionary.h

```diff
@@ -122,6 +123,71 @@ class AnyDictionary : private std::map<std::string, any>
         map::swap(other);
     }
 
+    /// @TODO: remove all of these @{
+
+    // if key is in this, and the type of key matches the type of result, then
+    // set result to the value of any_cast<type>(this[key]) and return true,
+    // otherwise return false
+    template<typename containedType>
+    bool
+    get_if_set(
+            const std::string& key,
+            containedType* result
+    ) const
+    {
+        const auto it = this->find(key);
+
+        if (
+                 (it != this->end())
+                 && (
+                     it->second.type().hash_code() 
+                     == typeid(containedType).hash_code()
+                )
+        ) 
+        {
+            *result = any_cast<containedType>(it->second);
```

Comment:
I'm not sure how to repro this case, but if I try this, I get a compile error:
```c++
    otio::AnyDictionary test;
    test["danger"] = nullptr;
    decltype(nullptr) result;
    test.get_if_set("danger", result);
```

Reply:
Could that type be blocked by some other means?

```c++
&& (
                    typeid(decltype(nullptr)).hash_code() 
                    != typeid(containedType).hash_code()
                )
```

Reply:
Sorry, I meant to say `result` not `containedType` (nullptr is not a type in an of itself). You could block it in the condition by just `result != nullptr` if you want to avoid changing the return value.

Reply:
got it, added a check